src/libdredd/src/mutation_replace_binary_operator.cc

```diff
@@ -69,171 +69,203 @@ bool MutationReplaceBinaryOperator::IsValidReplacementOperator(
             op == clang::BO_XorAssign));
 }
 
-std::string MutationReplaceBinaryOperator::GenerateMutatorFunction(
-    const std::string& function_name, const std::string& result_type,
-    const std::string& lhs_type, const std::string& rhs_type,
-    const std::vector<clang::BinaryOperatorKind>& operators,
-    int& mutation_id) const {
-  std::stringstream new_function;
-  new_function << "static " << result_type << " " << function_name;
-  new_function << "(std::function<" << lhs_type << "()>"
-               << " arg1, ";
-
-  new_function << "std::function<" << rhs_type << "()>"
-               << " arg2, int local_mutation_id) {\n";
-
-  int mutant_offset = 0;
+std::string MutationReplaceBinaryOperator::GetFunctionName(
+    clang::ASTContext& ast_context) const {
+  std::string lhs_type = binary_operator_.getLHS()
+                             ->getType()
+                             ->getAs<clang::BuiltinType>()
+                             ->getName(ast_context.getPrintingPolicy())
+                             .str();
+  std::string rhs_type = binary_operator_.getRHS()
+                             ->getType()
+                             ->getAs<clang::BuiltinType>()
+                             ->getName(ast_context.getPrintingPolicy())
+                             .str();
```

Comment:
Now that we don't need to use (these versions of) lhs_type and rhs_type elsewhere, I think it would make sense simply to build `function_name` up bit by bit. So first populate it with `__dredd_replace_binary_operator_` and then the operator name, and after that tack on the LHS and RHS types, and call `SpaceToUnderscore` on the fly as you tack them on.

Reply:
Done.